flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlReplaceTableAs.java

```diff
@@ -0,0 +1,278 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.sql.parser.ddl;
+
+import org.apache.flink.sql.parser.ExtendedSqlNode;
+import org.apache.flink.sql.parser.SqlConstraintValidator;
+import org.apache.flink.sql.parser.SqlUnparseUtils;
+import org.apache.flink.sql.parser.ddl.constraint.SqlTableConstraint;
+import org.apache.flink.sql.parser.error.SqlValidateException;
+
+import org.apache.calcite.sql.SqlCharStringLiteral;
+import org.apache.calcite.sql.SqlCreate;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlSpecialOperator;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+/** Replace table as statement. */
+public class SqlReplaceTableAs extends SqlCreate implements ExtendedSqlNode {
+
+    public static final SqlSpecialOperator OPERATOR =
+            new SqlSpecialOperator("REPLACE TABLE AS", SqlKind.OTHER_DDL);
+
+    private final SqlIdentifier tableName;
+
+    private final SqlNodeList columnList;
+
+    private final SqlNodeList propertyList;
+
+    private final List<SqlTableConstraint> tableConstraints;
+
+    private final SqlNodeList partitionKeyList;
+
+    private final SqlWatermark watermark;
+
+    private final SqlCharStringLiteral comment;
+
+    private final boolean isTemporary;
+
+    private final boolean isCreateOrReplace;
+
+    private final SqlNode asQuery;
+
+    public SqlReplaceTableAs(
+            SqlParserPos pos,
+            SqlIdentifier tableName,
+            SqlNodeList columnList,
+            List<SqlTableConstraint> tableConstraints,
+            SqlNodeList propertyList,
+            SqlNodeList partitionKeyList,
+            @Nullable SqlWatermark watermark,
+            @Nullable SqlCharStringLiteral comment,
+            SqlNode asQuery,
+            boolean isTemporary,
+            boolean ifNotExists,
+            boolean isCreateOrReplace) {
+        this(
+                OPERATOR,
+                pos,
+                tableName,
+                columnList,
+                tableConstraints,
+                propertyList,
+                partitionKeyList,
+                watermark,
+                comment,
+                asQuery,
+                isTemporary,
+                ifNotExists,
+                isCreateOrReplace);
+    }
+
+    public SqlReplaceTableAs(
```

Comment:
Do we really need this construct method? Can't we use the construct method above.